src/__tests__/actions-test.ts

```diff
@@ -30,92 +31,92 @@ describe("actions", () => {
 
       await actions.postForm(type, url, formData, "POST", "", "JSON")(dispatch);
+      let fetchArgs = fetchMock.calls();
+
       expect(dispatch.callCount).to.equal(3);
       expect(dispatch.args[0][0].type).to.equal(`${type}_${ActionCreator.REQUEST}`);
       expect(dispatch.args[1][0].type).to.equal(`${type}_${ActionCreator.SUCCESS}`);
       expect(dispatch.args[2][0].type).to.equal(`${type}_${ActionCreator.LOAD}`);
-      expect(dispatch.args[2][0].data).to.equal(textResponse);
-      expect(fetchMock.callCount).to.equal(1);
-      expect(fetchMock.args[0][0]).to.equal(url);
-      expect(fetchMock.args[0][1].method).to.equal("POST");
+      expect(dispatch.args[2][0].data.text).to.equal(textResponse);
+      expect(fetchMock.called()).to.equal(true);
+      expect(fetchArgs[0][0]).to.equal(url);
+      expect(fetchArgs[0][1].method).to.equal("POST");
       const expectedHeaders = new Headers();
-      expect(fetchMock.args[0][1].headers).to.deep.equal(expectedHeaders);
-      expect(fetchMock.args[0][1].body).to.equal(formData);
+      expect(fetchArgs[0][1].headers).to.deep.equal(expectedHeaders);
+      expect(fetchArgs[0][1].body).to.equal(formData);
     });
 
     it("dispatches a DELETE request", async () => {
       const dispatch = stub();
-      const responseText = stub().returns(new Promise<string>((resolve) => {
+      const responseText = new Promise<string>((resolve) => {
         resolve("response");
-      }));
-      const fetchMock = stub().returns(new Promise<any>((resolve, reject) => {
-        resolve({ status: 200, text: responseText });
-      }));
-      fetch = fetchMock;
+      });
+      fetchMock
+        .mock(url, responseText);
 
       await actions.postForm(type, url, formData, "DELETE")(dispatch);
+      let fetchArgs = fetchMock.calls();
+
       expect(dispatch.callCount).to.equal(3);
-      expect(fetchMock.callCount).to.equal(1);
-      expect(fetchMock.args[0][0]).to.equal(url);
-      expect(fetchMock.args[0][1].method).to.equal("DELETE");
+      expect(fetchMock.called()).to.equal(true);
+      expect(fetchArgs[0][0]).to.equal(url);
+      expect(fetchArgs[0][1].method).to.equal("DELETE");
       const expectedHeaders = new Headers();
-      expect(fetchMock.args[0][1].headers).to.deep.equal(expectedHeaders);
-      expect(fetchMock.args[0][1].body).to.equal(formData);
+      expect(fetchArgs[0][1].headers).to.deep.equal(expectedHeaders);
+      expect(fetchArgs[0][1].body).to.equal(formData);
     });
 
     it("dispatches failure on bad response", async () => {
       const dispatch = stub();
-      const jsonResponse = new Promise(resolve => resolve({ detail: "problem detail" }));
-      const fetchMock = stub().returns(new Promise<any>(resolve => {
-        resolve({ status: 500, json: () => jsonResponse });
-      }));
-      fetch = fetchMock;
+      fetchMock
+        .mock(url, { status: 500, body: { detail: "problem detail" } });
```

Comment:
What bothers me about this module is that you can't add any values in the `json` prop. Adding it in the body returns what is needed but it doesn't seem right to me. Following the if/else in the function is still the same as before.

Reply:
This seems fine to me. Before we had to mock the `json` method on the response since it wasn't a real response, but maybe the new library is creating a more realistic response already.